django/forms/formsets.py

```diff
@@ -325,8 +325,12 @@ def full_clean(self):
             # Empty forms are unchanged forms beyond those with initial data.
             if not form.has_changed() and i >= self.initial_form_count():
                 empty_forms_count += 1
-
-            self._errors.append(form.errors)
+            # Call form.errors first so we can get cleaned_data.
```

Comment:
I think a comment here makes sense, I wondered why you were performing validation on the form if it was meant to be deleted before reading it 👍 You might want to mention that it's `_should_delete_form` that requires `cleaned_data` and not _we_.